src/frame_decoder.rs

```diff
@@ -622,3 +629,73 @@ impl Read for FrameDecoder {
         }
     }
 }
+
+/// Decode multiple frames into the output slice.
+///
+/// `input` must contain an exact number of frames.
+///
+/// `output` must be large enough to hold the decompressed data. If you don't know
+/// how large the output will be, use [`FrameDecoder`] instead.
+///
+/// Returns the number of bytes written to `output`.
+pub fn decode_all(mut input: &[u8], mut output: &mut [u8]) -> Result<usize, FrameDecoderError> {
+    let mut total_bytes_written = 0;
+    while !input.is_empty() {
+        let mut decoder = FrameDecoder::new();
+        match decoder.init(&mut input) {
+            Ok(_) => {}
+            Err(FrameDecoderError::ReadFrameHeaderError(
+                frame::ReadFrameHeaderError::SkipFrame { length, .. },
+            )) => {
+                input = input
+                    .get(length as usize..)
+                    .ok_or(FrameDecoderError::FailedToSkipFrame)?;
+                continue;
+            }
+            Err(e) => return Err(e),
+        };
+        loop {
+            decoder.decode_blocks(&mut input, BlockDecodingStrategy::UptoBlocks(1))?;
```

Comment:
This strategy is different from what `StreamingDecoder` uses, but I chose it because I assume it will reduce memory usage without affecting performance.

Reply:
I think 1 block is a bit too restrictive, this means decoding in 128kb blocks, copying the results and continuing. I'll benchmark this to see how severe the impact is. Functionality wise this doesn't really matter so I won't block the PR ont his.

Reply:
Is there some overhead that is incurred from copying more often? What do you think would be a better size? I don't think we want to use `BlockDecodingStrategy::UptoBytes(output.len())` because if the input is a single large frame then that roughly doubles the total memory usage, and probably results in a few unwanted reallocs as the buffer grows.

Ideally the decoder would use `output` directly and avoid a copy in the first place, but the code doesn't seem to be set up to allow that.

Reply:
> Ideally the decoder would use output directly and avoid a copy in the first place, but the code doesn't seem to be set up to allow that.

That would be ideal in terms of memory usage but the algorithm always needs access to the last window-size bytes where the window size is defined by the frame header. You could spill data directly to the output but that would also mean *a lot* of small copies.

> Is there some overhead that is incurred from copying more often?

It just means that you enter the decoding loop more often to copy results in small chunks. It's not going to be a *huge* difference but I'm going to guess it's noticable.

> I don't think we want to use BlockDecodingStrategy::UptoBytes(output.len())

I agree. Looking at it now, I'd limit the streaming decoder to only decode to a reasonable limit like a few MB. The expectation of the StreamingDecoder wasn't that the output would be a huge buffer but I see how that can definitely be the case and be bad in terms of memory usage. This is what I'd recommend for this implementation too.

Reply:
> That would be ideal in terms of memory usage but the algorithm always needs access to the last window-size bytes where the window size is defined by the frame header anywas so some dynamic buffering will have to be done by the decoder.

I know you need that for streaming, but in this case it should be able to use `output` for the  window buffer.

Reply:
I guess that's true but I think it's pretty uncommon to know how large the decompressed frame is going to be. Maybe the Sequence Execution (currently implemented on a ringbuffer) could be made pluggable. It would remove the need for all the memcpy calls needed to collect bytes from the decoder which would be nice. Not sure it's worth the effort though tbh.